### DISCUSSION TITLE
Border router missing route?

### DISCUSSION THREAD

**Discussion by FHFS** [openthread/openthread#9313] [Q&A] [answered]
Hello,

I've got a local network with a Google Nesthub 2 as a Border Router.
When I deploy my nrf52840 onto the thread network everything works fine. I can call otDnsClientBrowse to get an address for the mDNS service, and connect to it on the local network.

I've tried to replace the Nesthub with esp-thread-br. When I do that I am able to connect to the local network with the nrf52840, but I get a timeout on otDnsClientBrowse with some debug lines saying "Failed to find valid route for" the mesh local address for the esp-thread-br.  And some dropped udp packets with bad checksums on the debug output of the esp-thread-br.

When I then add the Nesthub to the network the problem goes away. Even after I remove the Nesthub the problem remains solved until I flash erase the esp-thread-br.

I've checked the netdata in both situations and neither have any routes to the mesh local prefixes.
What did the Nesthub enable in the esp-thread-br that was disabled before?

Additional information - no route to mesh local:
```
netdata show
Prefixes:
fd05:1a01:67e7:1::/64 paos low f400
Routes:
2001:db8:1234:1::/64 s med f400
fd05:1a01:67e7:2:0:0::/96 sn low f400
Services:
44970 5d fddead00beef00008c88f0b757b89480d11f s f400

srp client server
[fdde:ad00:beef:0:8c88:f0b7:57b8:9480]:53535

ip print
netif: nt
nt inet6: FD05:1A01:67E7:2::1 48
netif: ot
ot inet6: FE80::8891:AD20:B5BF:DA03 48
ot inet6: FDDE:AD00:BEEF:0:8C88:F0B7:57B8:9480 16
ot inet6: FD05:1A01:67E7:1:F0BD:4AB6:F1FC:44BF 48
ot inet6: FF03::2
ot inet6: FF02::2
ot inet6: FF33:40:FDDE:AD00:BEEF::1
ot inet6: FF32:40:FDDE:AD00:BEEF::1
ot inet6: FF03::FC
ot inet6: FF03::1
ot inet6: FF02::1
netif: st
st inet6: FE80::3685:18FF:FE7E:7E8C 48
st inet6: 2001:DB8:1234:1:3685:18FF:FE7E:7E8C 48
st inet6: FF02::16
st inet6: FF02::2
st inet6: FF02::FB
st inet6: FF02::1:FF7E:7E8C
```
Error message nrf52840:
```
[00:00:03.765,930] <inf> ear_ot_mqtt: Default dns server address:
[00:00:03.766,143] <inf> ear_ot_mqtt: fdde:ad00:beef:0:8c88:f0b7:57b8:9480
[00:00:03.766,998] <inf> [N] Ip6-----------: Failed to find valid route for: fdde:ad00:beef:0:8c88:f0b7:57b8:9480
```
Error message esp-thread-br
```
I (3435914) OPENTHREAD: Received PIO
I(3436904) OPENTHREAD:[N] Ip6-----------: Error UDP Checksum
I(3436904) OPENTHREAD:[N] Ip6-----------: Failed to handle payload: Drop
```
Additional information - after adding and removing nesthub
```
Netdata show
Prefixes:
fd97:bf69:3545:1::/64 paos low 2c00
Routes:
2001:db8:1234:1::/64 s med 2c00
fd97:bf69:3545:2:0:0::/96 sn low 2c00
Services:
44970 5d fd8bbdd9ef0b0000f155de541c6ce7eed11f s 2c00

srp client server
[fd8b:bdd9:ef0b:0:f155:de54:1c6c:e7ee]:53535

ip print
netif: nt
nt inet6: FD97:BF69:3545:2::1 48
netif: ot
ot inet6: FE80::206F:C8FF:8CE6:8EE2 48
ot inet6: FD8B:BDD9:EF0B:0:F155:DE54:1C6C:E7EE 16
ot inet6: FD97:BF69:3545:1:83E9:C983:2ECF:E6B6 48
ot inet6: FF03::2
ot inet6: FF02::2
ot inet6: FF33:40:FD8B:BDD9:EF0B::1
ot inet6: FF32:40:FD8B:BDD9:EF0B::1
ot inet6: FF03::FC
ot inet6: FF03::1
ot inet6: FF02::1
netif: st
st inet6: FE80::3685:18FF:FE7E:7E8C 48
st inet6: 2001:DB8:1234:1:3685:18FF:FE7E:7E8C 48
st inet6: FF02::16
st inet6: FF02::2
st inet6: FF02::FB
st inet6: FF02::1:FF7E:7E8C
```

**Comment by jwhui** [comment, marked as answer]
Verify that the Active Operational Dataset between both the border routers are exactly the same using the `dataset active` CLI command.

The `FDDE:AD00:BEEF:0:8C88::/64` mesh-local prefix seems suspicious, since the mesh-local prefix should be randomly generated.

When forming a new Thread network, use the `dataset init new` command as described in [README_DATASET.md](https://github.com/openthread/openthread/blob/main/src/cli/README_DATASET.md#form-network).

**Reply by FHFS** [reply]
Exactly right!

When I do a `dataset init new` followed by setting the channel panid extpan networkname/key, everything works as expected.

I think the problem lies with the dataset initialization on the esp-thread-br then.
I've compiled the esp-thread-br project with `CONFIG_OPENTHREAD_NETWORK_` for channel, panid, extpan, networkname/key.

So the initialization of the dataset by `esp_openthread_auto_start` is not correct, is that a correct conclusion?

I will make a github issue on the esp-thread-br github to resolve this.

Thanks again @jwhui !
